### ISSUE TITLE
Google Sign In - Mac Catalyst support (iPad apps to Mac)

### ISSUE THREAD

**Issue by filippozanfini** [googlesamples/google-services#420]
Will Google Sign In be supported on iPad apps running on Mac?

**Comment by terrillt** [comment]
This is also a critical need for our project.

**Comment by LucaIaconelli** [comment]
Any news?

**Comment by chethanashastry** [comment]
This is required for our app. Is this in pipeline? Any news?

**Comment by appfrilans** [comment]
Is there any alternatives in the meantime?

**Comment by yilei** [comment]
Ping?

**Comment by girishw** [comment]
+1 This is critical for me.

**Comment by itayAmza** [comment]
Updates?

**Comment by girishw** [comment]
Any updates?

**Comment by ptsochantaris** [comment]
It would be great if there was some update to this issue along with a sense of some timeframe. 

Catalyst support is especially important for internal tools that use Google Auth on our end. At the very least perhaps some suggestions on alternative workarounds for the issue?

**Comment by sayantan300** [comment]
Who are you?

**Comment by trannguyenvu91** [comment]
Any updates?

**Comment by ptsochantaris** [comment]
Any updates? (he also asked, expecting full well the answer to be crickets)

Does anybody else have thoughts or experience on possible workarounds for this problem? Any info would be greatly appreciated.

**Comment by ondrejhanak** [comment]
I was able to make Google Sign In work with this: https://github.com/elsesiy/GAppAuth.

**Comment by ptsochantaris** [comment]
@ondrejhanak Indeed, GAppAuth worked great, many thanks for the pointer. Amazing that the official SDK remains broken to this day.

**Comment by natecraft1** [comment]
@ondrejhanak @ptsochantaris Would you mind posting a gist? Having trouble with this

**Comment by ptsochantaris** [comment]
@natecraft1 I personally just followed the iOS instructions on the README in that repo, and things worked great, _except_ one issue where the server needed me to add some extra auth contexts via `GAppAuth.shared.appendAuthorizationRealm`, in our case it needed:
```
        GAppAuth.shared.appendAuthorizationRealm("email")
        GAppAuth.shared.appendAuthorizationRealm("profile")
```
in order to get the same id token that the app used to send to the server when it used the Google SDK. I assume the official SDK probably adds them by default? Apart from that, it "just worked". I hope this is of some help!

**Comment by girishw** [comment]
> @ondrejhanak @ptsochantaris Would you mind posting a gist? Having trouble with this

What is the issue you are running into?

**Comment by natecraft1** [comment]
Chrome just redirects to google.com and the `GAppAuth.shared.authorize()` callback never gets called. If I change the default browser to Safari it says macOS doesn't recognize the internet address after trying to sign in with google.

**Comment by ApostolisApo** [comment]
> @natecraft1 I personally just followed the iOS instructions on the README in that repo, and things worked great, _except_ one issue where the server needed me to add some extra auth contexts via `GAppAuth.shared.appendAuthorizationRealm`, in our case it needed:
> 
> ```
>         GAppAuth.shared.appendAuthorizationRealm("email")
>         GAppAuth.shared.appendAuthorizationRealm("profile")
> ```
> 
> in order to get the same id token that the app used to send to the server when it used the Google SDK. I assume the official SDK probably adds them by default? Apart from that, it "just worked". I hope this is of some help!

how do you get the client secret from the firebase app on console?

**Comment by ApostolisApo** [comment]
> @natecraft1 I personally just followed the iOS instructions on the README in that repo, and things worked great, _except_ one issue where the server needed me to add some extra auth contexts via `GAppAuth.shared.appendAuthorizationRealm`, in our case it needed:
> 
> ```
>         GAppAuth.shared.appendAuthorizationRealm("email")
>         GAppAuth.shared.appendAuthorizationRealm("profile")
> ```
> 
> in order to get the same id token that the app used to send to the server when it used the Google SDK. I assume the official SDK probably adds them by default? Apart from that, it "just worked". I hope this is of some help!

after the authorize succeeds how do we sign in to Firebase? Can I use this? Auth.auth().signIn(withCustomToken: idToken)

**Comment by natecraft1** [comment]
In my ViewController: 
```
class ViewController: NSViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        do {
            try GAppAuth.shared.authorize { (success) in
                // never called
            }
        } catch {
            print(error)
        }
    }
```

What happens is that it pops Safari, and when I sign in with my email, it just lands me on google.com and the callback is not called. Could it be an issue with the redirect URI? Also, why it doesn't use the default browser? Is that configurable?

Thanks @ptsochantaris @girishw @ApostolisApo 🙌

**Comment by masaldana2** [comment]
I NEED THIS

**Comment by narek-sv** [comment]
Any updates, Google? What an unresponsive team!

**Comment by appfrilans** [comment]
I am also waiting for this!

**Comment by kenoButler** [comment]
It would be great if one of the Google devs could provide some sort of update or ETA on this.
